internal/api/grpc/idp/converter.go

```diff
@@ -65,21 +64,21 @@ func ExternalIDPViewToLoginPolicyLinkPb(link *iam_model.IDPProviderView) *idp_pb
 	}
 }
 
-func IDPsToUserLinkPb(res []*user_model.ExternalIDPView) []*idp_pb.IDPUserLink {
+func IDPUserLinksToPb(res []*query.UserIDPLink) []*idp_pb.IDPUserLink {
 	links := make([]*idp_pb.IDPUserLink, len(res))
 	for i, link := range res {
-		links[i] = ExternalIDPViewToUserLinkPb(link)
+		links[i] = IDPUserLinkToPb(link)
 	}
 	return links
 }
 
-func ExternalIDPViewToUserLinkPb(link *user_model.ExternalIDPView) *idp_pb.IDPUserLink {
+func IDPUserLinkToPb(link *query.UserIDPLink) *idp_pb.IDPUserLink {
 	return &idp_pb.IDPUserLink{
 		UserId:           link.UserID,
-		IdpId:            link.IDPConfigID,
+		IdpId:            link.IDPID,
 		IdpName:          link.IDPName,
-		ProvidedUserId:   link.ExternalUserID,
-		ProvidedUserName: link.UserDisplayName,
+		ProvidedUserId:   link.ProvidedUserID,
+		ProvidedUserName: link.ProvidedUsername,
 		//TODO: as soon as saml is implemented we need to switch here
 		//IdpType: IDPTypeToPb(link.Type),
```

Comment:
do we have the type now with the new projection?

Reply:
not now, will be fixed in https://github.com/caos/zitadel/pull/2767